Implements phone auth support for Admin node.js SDK.

bojeil-google opened the pull request:
Adds phoneNumber to UserRecord and UserInfo.
Implements new API for looking up users by phone number: getUserByPhoneNumber
Adds the ability to pass the phoneNumber property when creating a new user,
or updating an existing one via createUser and updateUser.
This includes the ability to delete the phone number for a specified user.
Defines the new related auth errors:
auth/invalid-phone-number and auth/phone-number-already-exists.

hiranya911 commented on test/unit/auth/auth-api-request.spec.ts:
```diff
@@ -534,6 +564,75 @@ describe('FirebaseAuthRequestHandler', () => {
+    it('should be fulfilled given a valid phoneNumber', () => {
+      const expectedResult = {
+        users : [
+          {phoneNumber: '+11234567890'},
```
Should we also add the `providerUserInfo` field to the expected response, and see if get parsed correctly?

bojeil-google replied:
I added a more realistic response and I test the expected response is returned from this operation which is enough for this API. This API returns a response that resolves with the same server response. UserRecord parses that content. I have tests for that in that file. auth.ts passes the response to UserRecord. I have a check for the whole flow in auth.spec.ts. I stub the response of getAccountInfoByPhoneNumber and then check that it is parsed correctly into the expected UserRecord.

hiranya911 commented:
I believe we also need to update `index.d.ts` right?

bojeil-google commented:
Thanks for catching the missing definitions in index.d.ts. I added those. I also added detailed response for getAccountInfoByPhoneNumber. More detailed checks on parsing is adding for getUserByPhoneNumber.

hiranya911 commented:
LGTM

hiranya911 commented on test/integration/auth.js:
```diff
@@ -33,23 +33,75 @@ function test(utils) {
+    ];
+    return Promise.all(promises)
+      .catch(function(error) {
+        utils.logFailure('setUp()', error);
```
Perhaps log this as `before()` to be clear?

bojeil-google replied:
Thanks for catching this mislabel. I originally had named this setUp() but then renamed it to before().

hiranya911 commented:
LGTM with one minor suggestion.
